File: service-java/{{#server.micronaut}}/src/main/java/{{package.value}}/{{#jsonlib.moshi}}/MoshiFactory.java
package {{package.value}};

import com.squareup.moshi.Moshi;
import io.micronaut.context.annotation.*;
import jakarta.inject.Singleton;
import {{package.value}}.json.CustomMoshiAdapters;

@Factory
public class MoshiFactory {
    @Singleton
    public Moshi getMoshi() {
        var moshiBuilder = new Moshi.Builder();
		CustomMoshiAdapters.setup(moshiBuilder);
        return moshiBuilder.build();
    }
}

File: service-kotlin/{{#server.micronaut}}/src/main/kotlin/{{package.value}}/{{#jsonlib.moshi}}/MoshiFactory.kt
package {{package.value}};

import com.squareup.moshi.Moshi
import io.micronaut.context.annotation.Factory
import jakarta.inject.Singleton
import {{package.value}}.json.setupMoshiAdapters

@Factory
class MoshiFactory {
    @Singleton
    fun getMoshi(): Moshi {
        val moshiBuilder = Moshi.Builder()
        setupMoshiAdapters(moshiBuilder)
        return moshiBuilder.build()
    }
}